### PR TITLE
Making C# Configuration.DefaultTelemetryEndpoint be able to be accessed in public.

### PR DESCRIPTION
**Pull request by daoshengmu**
I think it makes sense to do this adjustment when we need to give `channel` name after `endpoint` for initializing Glean in C#.

```
 GleanInstance.Initialize(
                applicationId: "org.mozilla.csharp.tests",
                applicationVersion: "1.0-test",
                uploadEnabled: true,
                configuration: new Configuration(Configuration.DefaultTelemetryEndpoint, "fxr"),
                dataDir: tempDataDir
 );
```

### CODE REVIEW

File: glean-core/csharp/Glean/Configuration/Configuration.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public sealed class Configuration
         /// <summary>
         /// The default server pings are sent to.
         /// </summary>
-        public const string DefaultTelemetryEndpoint = "https://incoming.telemetry.mozilla.org";
+        public static readonly string DefaultTelemetryEndpoint = "https://incoming.telemetry.mozilla.org";
```
Comment (Dexterp37) [inline glean-core/csharp/Glean/Configuration/Configuration.cs:14]: Why was this changed?

**Comment by daoshengmu** [comment]
I confirmed. We don't need to make this change :)
